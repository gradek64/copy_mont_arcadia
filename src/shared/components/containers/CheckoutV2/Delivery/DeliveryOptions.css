/** @define DeliveryOptions; weak */

.DeliveryOptions {
  & .DeliveryOption {
    margin-bottom: 10px;

    &-content {
      display: block;
      cursor: pointer;
      padding: 15px;
      border: 1px solid $checkout-divider-color;
      width: 100%;
      text-align: left;

      &--selected {
        border: 2px solid $checkout-delivery-method-button-selected-color;
      }
    }

    &-title {
      display: block;
      font-weight: $checkout-delivery-title-font-weight;
      margin-bottom: 10px;
    }

    &-description {
      font-weight: $checkout-delivery-option-font-weight;
      font-size: 12px;
    },
    &-additionalDescription {
      display: block;
      font-weight: $checkout-delivery-option-font-weight;
      font-size: 12px;
    }
  }
}
