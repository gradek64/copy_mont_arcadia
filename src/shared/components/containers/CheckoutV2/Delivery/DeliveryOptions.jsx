import React from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'

const buttonClassName = (isLocationSelected) => {
  return classnames('DeliveryOption-content', {
    'DeliveryOption-content--selected': isLocationSelected,
  })
}

const DeliveryOptions = (
  { deliveryLocations, onChangeDeliveryLocation },
  { l }
) => (
  <article className="DeliveryOptions" aria-label="Delivery Location Options">
    {deliveryLocations.map((location) => (
      <div
        className={`DeliveryOption DeliveryOption--${location.deliveryLocationType.toLowerCase()}`}
        key={location.deliveryLocationType}
      >
        <button
          className={buttonClassName(location.selected)}
          onClick={onChangeDeliveryLocation.bind(null, location)}
          disabled={!location.enabled}
        >
          <span className="DeliveryOption-title h3">{l(location.title)}</span>
          <span className="DeliveryOption-description">
            {l(location.description)}
          </span>
          <span className="DeliveryOption-additionalDescription">
            {l(location.additionalDescription)}
          </span>
        </button>
      </div>
    ))}
  </article>
)

DeliveryOptions.propTypes = {
  deliveryLocations: PropTypes.arrayOf(
    PropTypes.shape({
      deliveryLocationType: PropTypes.string.isRequired,
      selected: PropTypes.boolean,
      label: PropTypes.string,
      iconUrl: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.description,
      additionalDescription: PropTypes.string.description,
      enabled: PropTypes.boolean,
    })
  ),
  onChangeDeliveryLocation: PropTypes.func,
}

DeliveryOptions.contextTypes = {
  l: PropTypes.func,
}

export default DeliveryOptions
