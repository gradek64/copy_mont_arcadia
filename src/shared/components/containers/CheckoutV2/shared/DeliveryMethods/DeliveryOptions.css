/** @define DeliveryOptionsSelect */

.DeliveryOptionsSelect {
  display: flex;
  flex-flow: row nowrap;
  width: 90%;
    &-dayButtonContainer {
      display: flex;
      flex-direction: column;
      margin-bottom: 10px;
      width: 5.6%;
      cursor: pointer;
      align-items: center;
      font-size: 0.8em;
      min-width: 40px;
    }
    &-dateButton {
      padding: 8px 10px;
      text-align: center;
      background: #F8F8F8;
      border-radius: 50%;
      margin: 4px 0;
      font-weight: 600;

      &--selected {
        background: black;
        color: white;
      }
    }

    &-dayText {
      text-transform: uppercase;
    }

    &-monthText {
      font-size: 0.85em;
    }

    &-dayButtonContainer:nth-last-child(n + 4),
     &-dayButtonContainer:nth-last-child(n + 4) ~ &-dayButtonContainer {
      flex-basis: 25%;
    }
    &-dayButtonContainer:nth-last-child(n + 5),
     &-dayButtonContainer:nth-last-child(n + 5) ~ &-dayButtonContainer {
      flex-basis: 20%;
    }
    &-dayButtonContainer:nth-last-child(n + 6),
     &-dayButtonContainer:nth-last-child(n + 6) ~ &-dayButtonContainer {
      flex-basis: 16.6%;
    }
    &-dayButtonContainer:nth-last-child(n + 7),
     &-dayButtonContainer:nth-last-child(n + 7) ~ &-dayButtonContainer {
      flex-basis: 20%;
    }
}

@media $mobile {
  .DeliveryOptionsSelect {
    width: 100%;
    padding-left: 5px;
  }
}

@media only screen and (min-width: 992px) {
  .DeliveryOptionsSelect {
    width: 60%;
    &-dayButtonContainer {
      flex-basis: 25%;
    }
  }
}
