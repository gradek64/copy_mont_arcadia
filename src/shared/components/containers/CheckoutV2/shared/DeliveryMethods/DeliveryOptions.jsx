import PropTypes from 'prop-types'
import React from 'react'
import classNames from 'classnames'

// Qubit Wrapper
import QubitReact from 'qubit-react/wrapper'

const DeliveryOptions = ({
  deliveryType,
  deliveryOptions,
  onChange,
  className,
  onAccordionToggle,
}) => {
  return deliveryOptions.length ? (
    <QubitReact
      id="qubit-delivery-calender-view"
      onChange={onChange}
      deliveryOptions={deliveryOptions}
    >
      <div className={className}>
        {deliveryOptions &&
          deliveryOptions.map((option) => {
            const { dateText, dayText, selected, shipModeId } = option
            const buttonClassNames = classNames(`${className}-dateButton`, {
              [`${className}-dateButton--selected`]: selected,
            })
            const dayTextClassNames = classNames(`${className}-dayText`, {
              [`${className}-dayText--notSelected`]: !selected,
            })
            const monthTextClassNames = classNames(`${className}-monthText`, {
              [`${className}-monthText--notSelected`]: !selected,
            })
            const dayButtonContainerClassNames = classNames(
              `${className}-dayButtonContainer`
            )
            return (
              <div key={dayText} className={dayButtonContainerClassNames}>
                <span className={dayTextClassNames}>{dayText && dayText}</span>
                <button
                  className={buttonClassNames}
                  name={deliveryType.toLowerCase()}
                  value={shipModeId}
                  onClick={({ target }) => {
                    onChange(target.value)
                    onAccordionToggle()
                  }}
                >
                  {dateText && dateText.substring(0, 2)}
                </button>
                <span className={monthTextClassNames}>
                  {dayText && dateText.substring(3, dateText.length)}
                </span>
              </div>
            )
          })}
      </div>
    </QubitReact>
  ) : null
}

DeliveryOptions.propTypes = {
  deliveryType: PropTypes.string.isRequired,
  deliveryOptions: PropTypes.arrayOf(
    PropTypes.shape({
      shipModeId: PropTypes.number,
      dayText: PropTypes.string,
      dateText: PropTypes.string,
      price: PropTypes.string,
      selected: PropTypes.bool,
    })
  ),
  onChange: PropTypes.func,
}

DeliveryOptions.defaultProps = {
  deliveryOptions: [],
  onChange: () => {},
}

DeliveryOptions.contextTypes = {
  p: PropTypes.func,
}

export default DeliveryOptions
