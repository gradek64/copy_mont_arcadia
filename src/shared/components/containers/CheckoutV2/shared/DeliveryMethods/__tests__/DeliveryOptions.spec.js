import testComponentHelper from 'test/unit/helpers/test-component'

import DeliveryOptions from '../DeliveryOptions'

describe('<DeliveryOptions />', () => {
  const renderComponent = testComponentHelper(DeliveryOptions)

  const createDeliveryOptions = ({ selectedShipModeId } = {}) => [
    {
      shipModeId: 28005,
      dayText: 'Thu',
      dateText: '07 Sep',
      price: '6.00',
      selected: selectedShipModeId === 28005,
    },
    {
      shipModeId: 28006,
      dayText: 'Fri',
      dateText: '08 Sep',
      price: '6.00',
      selected: selectedShipModeId === 28006,
    },
  ]

  describe('@render', () => {
    it('should render default state', () => {
      const component = renderComponent({
        deliveryType: 'HOME_STANDARD',
      })
      expect(component.getTree()).toMatchSnapshot()
    })

    it('should pass options to date select buttons', () => {
      const component = renderComponent({
        deliveryType: 'HOME_STANDARD',
        deliveryOptions: createDeliveryOptions({ selectedShipModeId: 28005 }),
      })
      expect(component.getTree()).toMatchSnapshot()
    })

    it('should pass value to first button as option selected', () => {
      const component = renderComponent({
        deliveryType: 'HOME_STANDARD',
        deliveryOptions: createDeliveryOptions({ selectedShipModeId: 28005 }),
        className: 'DeliveryOptionsSelect',
      })
      expect(component.wrapper.find('.DeliveryOptionsSelect-dateButton--selected').prop('value')).toBe(28005)
    })

    it('should handle Free Deliver', () => {
      const createDelivOptions = createDeliveryOptions()
      const component = renderComponent({
        deliveryType: 'HOME_STANDARD',
        deliveryCost: '0.00',
        deliveryOptions: createDelivOptions.map((option) => ({
          ...option,
          price: '0.00',
        })),
      })
      component.wrapper
        .find('button')
        .forEach((button, index) => {
          const propOption = component.wrapper.props().deliveryOptions
          expect(button.props().value).toBe(propOption[index].shipModeId)
          expect(propOption[index].price).toBe('0.00')
        })
    })
  })

  describe('@events', () => {
    it('should call `onChange` with targetâ€˜s value on change and call onAccordionToggle', () => {
      const onChangeMock = jest.fn()
      const onAccordionToggleMock = jest.fn()
      const component = renderComponent({
        deliveryType: 'HOME_STANDARD',
        deliveryOptions: createDeliveryOptions(),
        onChange: onChangeMock,
        onAccordionToggle: onAccordionToggleMock,
      })
      component.wrapper.find('button').first().prop('onClick')({
        target: { value: '28005' },
      })
      expect(onChangeMock).toHaveBeenCalledWith('28005')
      expect(onAccordionToggleMock).toHaveBeenCalled()
    })
  })
})
