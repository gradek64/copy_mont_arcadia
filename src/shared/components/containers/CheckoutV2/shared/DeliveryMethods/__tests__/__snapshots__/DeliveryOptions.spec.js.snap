// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeliveryOptions /> @render should pass options to date select buttons 1`] = `
<Component
  deliveryOptions={
    Array [
      Object {
        "dateText": "07 Sep",
        "dayText": "Thu",
        "price": "6.00",
        "selected": true,
        "shipModeId": 28005,
      },
      Object {
        "dateText": "08 Sep",
        "dayText": "Fri",
        "price": "6.00",
        "selected": false,
        "shipModeId": 28006,
      },
    ]
  }
  id="qubit-delivery-calender-view"
  onChange={[Function]}
>
  <div>
    <div
      className="undefined-dayButtonContainer"
      key="Thu"
    >
      <span
        className="undefined-dayText"
      >
        Thu
      </span>
      <button
        className="undefined-dateButton undefined-dateButton--selected"
        name="home_standard"
        onClick={[Function]}
        value={28005}
      >
        07
      </button>
      <span
        className="undefined-monthText"
      >
        Sep
      </span>
    </div>
    <div
      className="undefined-dayButtonContainer"
      key="Fri"
    >
      <span
        className="undefined-dayText undefined-dayText--notSelected"
      >
        Fri
      </span>
      <button
        className="undefined-dateButton"
        name="home_standard"
        onClick={[Function]}
        value={28006}
      >
        08
      </button>
      <span
        className="undefined-monthText undefined-monthText--notSelected"
      >
        Sep
      </span>
    </div>
  </div>
</Component>
`;

exports[`<DeliveryOptions /> @render should render default state 1`] = `""`;
